app/hooks/use-price-conversion.ts

```diff
@@ -0,0 +1,69 @@
+import { usePriceContext } from "@app/store/price-context"
+import { PaymentAmount, WalletCurrency } from "@app/types/amounts"
+import * as React from "react"
+
+export const usePriceConversion = () => {
+  const priceData = usePriceContext()
+  const convertCurrencyAmount = React.useCallback(
+    ({ amount, from, to }) => {
+      if (!priceData.initialized) {
+        return NaN
+      }
+      if (from === "BTC" && to === "USD") {
+        return (amount * priceData.price) / 100
+      }
+      if (from === "USD" && to === "BTC") {
+        return (100 * amount) / priceData.price
+      }
+      return amount
+    },
+    [priceData],
+  )
+
+  const convertPaymentAmount = React.useCallback(
+    (paymentAmount, toCurrency: WalletCurrency) => {
```

Comment:
Same here - I know this isn't your code but could we add types for args and return value for this function?